File: app/src/main/java/com/maufonseca/mauriciocv/presentation/main/SearchPresenter.java
package com.maufonseca.mauriciocv.presentation.main;

import com.maufonseca.mauriciocv.model.Snippet;

import java.util.ArrayList;

public class SearchPresenter {

  MainActivity activity;

  public SearchPresenter(MainActivity activity) {
    this.activity = activity;
  }

  public void onSearchResults(ArrayList<Snippet> snippets) {
    activity.updateResults(snippets);
  }
}

File: app/src/main/java/com/maufonseca/mauriciocv/presentation/main/SearchPresenter.kt
package com.maufonseca.mauriciocv.presentation.main

import com.maufonseca.mauriciocv.model.Snippet

import java.util.ArrayList

class SearchPresenter(var activity: SearchDisplay) : SearchPresentation {

  override fun onSearchResults(snippets: ArrayList<Snippet>) {
    activity.updateResults(snippets)
  }
}
